Trakt/Configuration/configPage.html

```diff
@@ -155,48 +173,88 @@ <h3 class="checkboxListLabel">Exclude folders:</h3>
+                ApiClient.getPluginConfiguration(TraktConfigurationPage.pluginUniqueId).then(function (config) {
+                    var currentUserConfig = config.TraktUsers.filter(function (curr) {
+                        return curr.LinkedMbUserId == currentUserId;
+                    })[0];
+                    // User doesn't have a config, so create a default one.
+                    if (!currentUserConfig) {
+                        currentUserConfig = {};
+                        config.TraktUsers.push(currentUserConfig);
+                    }
+                    currentUserConfig.SkipUnwatchedImportFromTrakt = $('#chkSkipUnwatchedImportFromTrakt', page).checked();
+                    currentUserConfig.PostWatchedHistory = $('#chkPostWatchedHistory', page).checked();
+                    currentUserConfig.ExtraLogging = $('#chkExtraLogging', page).checked();
+                    currentUserConfig.ExportMediaInfo = $('#chkExportMediaInfo', page).checked();
+                    currentUserConfig.LinkedMbUserId = currentUserId;
+                    currentUserConfig.LocationsExcluded = $('.chkTraktLocation:checked', page).map(function () {
+                        return this.getAttribute('data-location');
+                    }).get();
+                    if (currentUserConfig.UserName == '') {
+                        config.TraktUsers.remove(config.TraktUsers.indexOf(currentUserConfig));
+                    }
+                    ApiClient.updatePluginConfiguration(TraktConfigurationPage.pluginUniqueId, config).then(function (result) {
+                        Dashboard.processPluginConfigurationUpdateResult(result);
+                        ApiClient.getUsers().then(function (users) {
+                            var currentUserId = $('#selectUser', page).val();
+                            TraktConfigurationPage.populateUsers(users);
+                            $('#selectUser', page).val(currentUserId);
+                            TraktConfigurationPage.loadConfiguration(currentUserId, page);
+                            Dashboard.alert('Settings saved.');
+                            dfd.resolve();
                         });
                     });
+                });
+                return dfd.promise();
+            }
+            
+            $('.traktConfigurationPage').on('pageinit', function () {
+                Dashboard.showLoadingMsg();
+                var page = this;
+                $('#traktConfigurationForm', page).on('submit', function(){
+                    save(page);
                     return false;
                 });
+                $('#authorizeDevice', page).on('click', function () {
+                    var currentUserId = $('#selectUser').val();
+                    var headers = {
+                        accept: "application/json"
+                    };
+                    var request = {
+                        url: '/Trakt/Users/' + currentUserId + '/Authorize',
```

Comment:
Can we make relative URL here? This should help making prefixed deployments (behind some reverse proxies) work better.

Reply:
Fixed